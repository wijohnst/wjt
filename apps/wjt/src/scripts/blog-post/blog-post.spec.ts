import { BlogPost, getImageNodes } from '../blog-post';
import {
  rawPostMocks,
  rawContentMocks,
  getMockFrontmatter,
} from './blog-post.mocks';

describe('blog-post', () => {
  describe('BlogPost', () => {
    test('should be defined', () => {
      expect(BlogPost).toBeDefined();
    });
  });

  let sut: BlogPost;

  beforeEach(() => {
    sut = new BlogPost(rawPostMocks[0]);
  });

  test('should return a parsed post object', () => {
    const frontMatter = getMockFrontmatter(
      'Post 1',
      'Some Author',
      'post-1',
      'This is a sample mock post used for testing.'
    );
    const content = rawContentMocks[0];

    expect(sut.parsedPost).toStrictEqual({
      frontMatter,
      content,
    });
  });

  test('should return the rendered post', () => {
    expect(sut.postMarkup).toMatchSnapshot();
  });

  test('should return an array of post images', () => {
    expect(sut.postImages).toHaveLength(1);
    expect(sut.postImages).toMatchSnapshot();
  });

  describe('blogImages', () => {
    test('should be defined', () => {
      expect(sut.blogImages).toBeDefined();
    });

    test('should return an array of blog images', () => {
      expect(sut.blogImages).toHaveLength(1);
      expect(sut.blogImages).toMatchSnapshot();
    });
  });

  test('should update image sources', () => {
    sut.updateImageSources([
      {
        originalSrc: './path/to/image-1.jpg',
        newSrc: 'https://example.com/path/to/image-1.jpg',
      },
    ]);

    const imageNodes = getImageNodes(sut.parsedPost.content);

    expect(imageNodes[0].destination).toBe(
      'https://example.com/path/to/image-1.jpg'
    );
    expect(sut.postMarkup).toMatchSnapshot();
  });
});
