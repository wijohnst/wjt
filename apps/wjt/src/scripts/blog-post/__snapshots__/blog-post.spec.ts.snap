// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blog-post should return an array of post images 1`] = `
[
  {
    "altText": "image$100x200",
    "cdnEndpoint": undefined,
    "imageNode": Node {
      "_destination": "./path/to/image-1.jpg",
      "_fenceChar": null,
      "_fenceLength": 0,
      "_fenceOffset": null,
      "_firstChild": Node {
        "_destination": null,
        "_fenceChar": null,
        "_fenceLength": 0,
        "_fenceOffset": null,
        "_firstChild": null,
        "_info": null,
        "_isFenced": false,
        "_lastChild": null,
        "_level": null,
        "_listData": {},
        "_literal": "image$100x200",
        "_next": null,
        "_onEnter": null,
        "_onExit": null,
        "_open": true,
        "_parent": [Circular],
        "_prev": null,
        "_sourcepos": undefined,
        "_string_content": null,
        "_title": null,
        "_type": "text",
      },
      "_info": null,
      "_isFenced": false,
      "_lastChild": Node {
        "_destination": null,
        "_fenceChar": null,
        "_fenceLength": 0,
        "_fenceOffset": null,
        "_firstChild": null,
        "_info": null,
        "_isFenced": false,
        "_lastChild": null,
        "_level": null,
        "_listData": {},
        "_literal": "image$100x200",
        "_next": null,
        "_onEnter": null,
        "_onExit": null,
        "_open": true,
        "_parent": [Circular],
        "_prev": null,
        "_sourcepos": undefined,
        "_string_content": null,
        "_title": null,
        "_type": "text",
      },
      "_level": null,
      "_listData": {},
      "_literal": null,
      "_next": null,
      "_onEnter": null,
      "_onExit": null,
      "_open": true,
      "_parent": Node {
        "_destination": null,
        "_fenceChar": null,
        "_fenceLength": 0,
        "_fenceOffset": null,
        "_firstChild": Node {
          "_destination": null,
          "_fenceChar": null,
          "_fenceLength": 0,
          "_fenceOffset": null,
          "_firstChild": null,
          "_info": null,
          "_isFenced": false,
          "_lastChild": null,
          "_level": null,
          "_listData": {},
          "_literal": "This is the first post.",
          "_next": Node {
            "_destination": null,
            "_fenceChar": null,
            "_fenceLength": 0,
            "_fenceOffset": null,
            "_firstChild": null,
            "_info": null,
            "_isFenced": false,
            "_lastChild": null,
            "_level": null,
            "_listData": {},
            "_literal": null,
            "_next": [Circular],
            "_onEnter": null,
            "_onExit": null,
            "_open": true,
            "_parent": [Circular],
            "_prev": [Circular],
            "_sourcepos": undefined,
            "_string_content": null,
            "_title": null,
            "_type": "softbreak",
          },
          "_onEnter": null,
          "_onExit": null,
          "_open": true,
          "_parent": [Circular],
          "_prev": null,
          "_sourcepos": undefined,
          "_string_content": null,
          "_title": null,
          "_type": "text",
        },
        "_info": null,
        "_isFenced": false,
        "_lastChild": [Circular],
        "_level": null,
        "_listData": {},
        "_literal": null,
        "_next": null,
        "_onEnter": null,
        "_onExit": null,
        "_open": false,
        "_parent": Node {
          "_destination": null,
          "_fenceChar": null,
          "_fenceLength": 0,
          "_fenceOffset": null,
          "_firstChild": Node {
            "_destination": null,
            "_fenceChar": null,
            "_fenceLength": 0,
            "_fenceOffset": null,
            "_firstChild": Node {
              "_destination": null,
              "_fenceChar": null,
              "_fenceLength": 0,
              "_fenceOffset": null,
              "_firstChild": null,
              "_info": null,
              "_isFenced": false,
              "_lastChild": null,
              "_level": null,
              "_listData": {},
              "_literal": "Post 1",
              "_next": null,
              "_onEnter": null,
              "_onExit": null,
              "_open": true,
              "_parent": [Circular],
              "_prev": null,
              "_sourcepos": undefined,
              "_string_content": null,
              "_title": null,
              "_type": "text",
            },
            "_info": null,
            "_isFenced": false,
            "_lastChild": Node {
              "_destination": null,
              "_fenceChar": null,
              "_fenceLength": 0,
              "_fenceOffset": null,
              "_firstChild": null,
              "_info": null,
              "_isFenced": false,
              "_lastChild": null,
              "_level": null,
              "_listData": {},
              "_literal": "Post 1",
              "_next": null,
              "_onEnter": null,
              "_onExit": null,
              "_open": true,
              "_parent": [Circular],
              "_prev": null,
              "_sourcepos": undefined,
              "_string_content": null,
              "_title": null,
              "_type": "text",
            },
            "_level": 1,
            "_listData": {},
            "_literal": null,
            "_next": [Circular],
            "_onEnter": null,
            "_onExit": null,
            "_open": false,
            "_parent": [Circular],
            "_prev": null,
            "_sourcepos": [
              [
                1,
                1,
              ],
              [
                1,
                8,
              ],
            ],
            "_string_content": null,
            "_title": null,
            "_type": "heading",
          },
          "_info": null,
          "_isFenced": false,
          "_lastChild": [Circular],
          "_level": null,
          "_listData": {},
          "_literal": null,
          "_next": null,
          "_onEnter": null,
          "_onExit": null,
          "_open": false,
          "_parent": null,
          "_prev": null,
          "_sourcepos": [
            [
              1,
              1,
            ],
            [
              4,
              39,
            ],
          ],
          "_string_content": null,
          "_title": null,
          "_type": "document",
        },
        "_prev": Node {
          "_destination": null,
          "_fenceChar": null,
          "_fenceLength": 0,
          "_fenceOffset": null,
          "_firstChild": Node {
            "_destination": null,
            "_fenceChar": null,
            "_fenceLength": 0,
            "_fenceOffset": null,
            "_firstChild": null,
            "_info": null,
            "_isFenced": false,
            "_lastChild": null,
            "_level": null,
            "_listData": {},
            "_literal": "Post 1",
            "_next": null,
            "_onEnter": null,
            "_onExit": null,
            "_open": true,
            "_parent": [Circular],
            "_prev": null,
            "_sourcepos": undefined,
            "_string_content": null,
            "_title": null,
            "_type": "text",
          },
          "_info": null,
          "_isFenced": false,
          "_lastChild": Node {
            "_destination": null,
            "_fenceChar": null,
            "_fenceLength": 0,
            "_fenceOffset": null,
            "_firstChild": null,
            "_info": null,
            "_isFenced": false,
            "_lastChild": null,
            "_level": null,
            "_listData": {},
            "_literal": "Post 1",
            "_next": null,
            "_onEnter": null,
            "_onExit": null,
            "_open": true,
            "_parent": [Circular],
            "_prev": null,
            "_sourcepos": undefined,
            "_string_content": null,
            "_title": null,
            "_type": "text",
          },
          "_level": 1,
          "_listData": {},
          "_literal": null,
          "_next": [Circular],
          "_onEnter": null,
          "_onExit": null,
          "_open": false,
          "_parent": Node {
            "_destination": null,
            "_fenceChar": null,
            "_fenceLength": 0,
            "_fenceOffset": null,
            "_firstChild": [Circular],
            "_info": null,
            "_isFenced": false,
            "_lastChild": [Circular],
            "_level": null,
            "_listData": {},
            "_literal": null,
            "_next": null,
            "_onEnter": null,
            "_onExit": null,
            "_open": false,
            "_parent": null,
            "_prev": null,
            "_sourcepos": [
              [
                1,
                1,
              ],
              [
                4,
                39,
              ],
            ],
            "_string_content": null,
            "_title": null,
            "_type": "document",
          },
          "_prev": null,
          "_sourcepos": [
            [
              1,
              1,
            ],
            [
              1,
              8,
            ],
          ],
          "_string_content": null,
          "_title": null,
          "_type": "heading",
        },
        "_sourcepos": [
          [
            3,
            1,
          ],
          [
            4,
            39,
          ],
        ],
        "_string_content": null,
        "_title": null,
        "_type": "paragraph",
      },
      "_prev": Node {
        "_destination": null,
        "_fenceChar": null,
        "_fenceLength": 0,
        "_fenceOffset": null,
        "_firstChild": null,
        "_info": null,
        "_isFenced": false,
        "_lastChild": null,
        "_level": null,
        "_listData": {},
        "_literal": null,
        "_next": [Circular],
        "_onEnter": null,
        "_onExit": null,
        "_open": true,
        "_parent": Node {
          "_destination": null,
          "_fenceChar": null,
          "_fenceLength": 0,
          "_fenceOffset": null,
          "_firstChild": Node {
            "_destination": null,
            "_fenceChar": null,
            "_fenceLength": 0,
            "_fenceOffset": null,
            "_firstChild": null,
            "_info": null,
            "_isFenced": false,
            "_lastChild": null,
            "_level": null,
            "_listData": {},
            "_literal": "This is the first post.",
            "_next": [Circular],
            "_onEnter": null,
            "_onExit": null,
            "_open": true,
            "_parent": [Circular],
            "_prev": null,
            "_sourcepos": undefined,
            "_string_content": null,
            "_title": null,
            "_type": "text",
          },
          "_info": null,
          "_isFenced": false,
          "_lastChild": [Circular],
          "_level": null,
          "_listData": {},
          "_literal": null,
          "_next": null,
          "_onEnter": null,
          "_onExit": null,
          "_open": false,
          "_parent": Node {
            "_destination": null,
            "_fenceChar": null,
            "_fenceLength": 0,
            "_fenceOffset": null,
            "_firstChild": Node {
              "_destination": null,
              "_fenceChar": null,
              "_fenceLength": 0,
              "_fenceOffset": null,
              "_firstChild": Node {
                "_destination": null,
                "_fenceChar": null,
                "_fenceLength": 0,
                "_fenceOffset": null,
                "_firstChild": null,
                "_info": null,
                "_isFenced": false,
                "_lastChild": null,
                "_level": null,
                "_listData": {},
                "_literal": "Post 1",
                "_next": null,
                "_onEnter": null,
                "_onExit": null,
                "_open": true,
                "_parent": [Circular],
                "_prev": null,
                "_sourcepos": undefined,
                "_string_content": null,
                "_title": null,
                "_type": "text",
              },
              "_info": null,
              "_isFenced": false,
              "_lastChild": Node {
                "_destination": null,
                "_fenceChar": null,
                "_fenceLength": 0,
                "_fenceOffset": null,
                "_firstChild": null,
                "_info": null,
                "_isFenced": false,
                "_lastChild": null,
                "_level": null,
                "_listData": {},
                "_literal": "Post 1",
                "_next": null,
                "_onEnter": null,
                "_onExit": null,
                "_open": true,
                "_parent": [Circular],
                "_prev": null,
                "_sourcepos": undefined,
                "_string_content": null,
                "_title": null,
                "_type": "text",
              },
              "_level": 1,
              "_listData": {},
              "_literal": null,
              "_next": [Circular],
              "_onEnter": null,
              "_onExit": null,
              "_open": false,
              "_parent": [Circular],
              "_prev": null,
              "_sourcepos": [
                [
                  1,
                  1,
                ],
                [
                  1,
                  8,
                ],
              ],
              "_string_content": null,
              "_title": null,
              "_type": "heading",
            },
            "_info": null,
            "_isFenced": false,
            "_lastChild": [Circular],
            "_level": null,
            "_listData": {},
            "_literal": null,
            "_next": null,
            "_onEnter": null,
            "_onExit": null,
            "_open": false,
            "_parent": null,
            "_prev": null,
            "_sourcepos": [
              [
                1,
                1,
              ],
              [
                4,
                39,
              ],
            ],
            "_string_content": null,
            "_title": null,
            "_type": "document",
          },
          "_prev": Node {
            "_destination": null,
            "_fenceChar": null,
            "_fenceLength": 0,
            "_fenceOffset": null,
            "_firstChild": Node {
              "_destination": null,
              "_fenceChar": null,
              "_fenceLength": 0,
              "_fenceOffset": null,
              "_firstChild": null,
              "_info": null,
              "_isFenced": false,
              "_lastChild": null,
              "_level": null,
              "_listData": {},
              "_literal": "Post 1",
              "_next": null,
              "_onEnter": null,
              "_onExit": null,
              "_open": true,
              "_parent": [Circular],
              "_prev": null,
              "_sourcepos": undefined,
              "_string_content": null,
              "_title": null,
              "_type": "text",
            },
            "_info": null,
            "_isFenced": false,
            "_lastChild": Node {
              "_destination": null,
              "_fenceChar": null,
              "_fenceLength": 0,
              "_fenceOffset": null,
              "_firstChild": null,
              "_info": null,
              "_isFenced": false,
              "_lastChild": null,
              "_level": null,
              "_listData": {},
              "_literal": "Post 1",
              "_next": null,
              "_onEnter": null,
              "_onExit": null,
              "_open": true,
              "_parent": [Circular],
              "_prev": null,
              "_sourcepos": undefined,
              "_string_content": null,
              "_title": null,
              "_type": "text",
            },
            "_level": 1,
            "_listData": {},
            "_literal": null,
            "_next": [Circular],
            "_onEnter": null,
            "_onExit": null,
            "_open": false,
            "_parent": Node {
              "_destination": null,
              "_fenceChar": null,
              "_fenceLength": 0,
              "_fenceOffset": null,
              "_firstChild": [Circular],
              "_info": null,
              "_isFenced": false,
              "_lastChild": [Circular],
              "_level": null,
              "_listData": {},
              "_literal": null,
              "_next": null,
              "_onEnter": null,
              "_onExit": null,
              "_open": false,
              "_parent": null,
              "_prev": null,
              "_sourcepos": [
                [
                  1,
                  1,
                ],
                [
                  4,
                  39,
                ],
              ],
              "_string_content": null,
              "_title": null,
              "_type": "document",
            },
            "_prev": null,
            "_sourcepos": [
              [
                1,
                1,
              ],
              [
                1,
                8,
              ],
            ],
            "_string_content": null,
            "_title": null,
            "_type": "heading",
          },
          "_sourcepos": [
            [
              3,
              1,
            ],
            [
              4,
              39,
            ],
          ],
          "_string_content": null,
          "_title": null,
          "_type": "paragraph",
        },
        "_prev": Node {
          "_destination": null,
          "_fenceChar": null,
          "_fenceLength": 0,
          "_fenceOffset": null,
          "_firstChild": null,
          "_info": null,
          "_isFenced": false,
          "_lastChild": null,
          "_level": null,
          "_listData": {},
          "_literal": "This is the first post.",
          "_next": [Circular],
          "_onEnter": null,
          "_onExit": null,
          "_open": true,
          "_parent": Node {
            "_destination": null,
            "_fenceChar": null,
            "_fenceLength": 0,
            "_fenceOffset": null,
            "_firstChild": [Circular],
            "_info": null,
            "_isFenced": false,
            "_lastChild": [Circular],
            "_level": null,
            "_listData": {},
            "_literal": null,
            "_next": null,
            "_onEnter": null,
            "_onExit": null,
            "_open": false,
            "_parent": Node {
              "_destination": null,
              "_fenceChar": null,
              "_fenceLength": 0,
              "_fenceOffset": null,
              "_firstChild": Node {
                "_destination": null,
                "_fenceChar": null,
                "_fenceLength": 0,
                "_fenceOffset": null,
                "_firstChild": Node {
                  "_destination": null,
                  "_fenceChar": null,
                  "_fenceLength": 0,
                  "_fenceOffset": null,
                  "_firstChild": null,
                  "_info": null,
                  "_isFenced": false,
                  "_lastChild": null,
                  "_level": null,
                  "_listData": {},
                  "_literal": "Post 1",
                  "_next": null,
                  "_onEnter": null,
                  "_onExit": null,
                  "_open": true,
                  "_parent": [Circular],
                  "_prev": null,
                  "_sourcepos": undefined,
                  "_string_content": null,
                  "_title": null,
                  "_type": "text",
                },
                "_info": null,
                "_isFenced": false,
                "_lastChild": Node {
                  "_destination": null,
                  "_fenceChar": null,
                  "_fenceLength": 0,
                  "_fenceOffset": null,
                  "_firstChild": null,
                  "_info": null,
                  "_isFenced": false,
                  "_lastChild": null,
                  "_level": null,
                  "_listData": {},
                  "_literal": "Post 1",
                  "_next": null,
                  "_onEnter": null,
                  "_onExit": null,
                  "_open": true,
                  "_parent": [Circular],
                  "_prev": null,
                  "_sourcepos": undefined,
                  "_string_content": null,
                  "_title": null,
                  "_type": "text",
                },
                "_level": 1,
                "_listData": {},
                "_literal": null,
                "_next": [Circular],
                "_onEnter": null,
                "_onExit": null,
                "_open": false,
                "_parent": [Circular],
                "_prev": null,
                "_sourcepos": [
                  [
                    1,
                    1,
                  ],
                  [
                    1,
                    8,
                  ],
                ],
                "_string_content": null,
                "_title": null,
                "_type": "heading",
              },
              "_info": null,
              "_isFenced": false,
              "_lastChild": [Circular],
              "_level": null,
              "_listData": {},
              "_literal": null,
              "_next": null,
              "_onEnter": null,
              "_onExit": null,
              "_open": false,
              "_parent": null,
              "_prev": null,
              "_sourcepos": [
                [
                  1,
                  1,
                ],
                [
                  4,
                  39,
                ],
              ],
              "_string_content": null,
              "_title": null,
              "_type": "document",
            },
            "_prev": Node {
              "_destination": null,
              "_fenceChar": null,
              "_fenceLength": 0,
              "_fenceOffset": null,
              "_firstChild": Node {
                "_destination": null,
                "_fenceChar": null,
                "_fenceLength": 0,
                "_fenceOffset": null,
                "_firstChild": null,
                "_info": null,
                "_isFenced": false,
                "_lastChild": null,
                "_level": null,
                "_listData": {},
                "_literal": "Post 1",
                "_next": null,
                "_onEnter": null,
                "_onExit": null,
                "_open": true,
                "_parent": [Circular],
                "_prev": null,
                "_sourcepos": undefined,
                "_string_content": null,
                "_title": null,
                "_type": "text",
              },
              "_info": null,
              "_isFenced": false,
              "_lastChild": Node {
                "_destination": null,
                "_fenceChar": null,
                "_fenceLength": 0,
                "_fenceOffset": null,
                "_firstChild": null,
                "_info": null,
                "_isFenced": false,
                "_lastChild": null,
                "_level": null,
                "_listData": {},
                "_literal": "Post 1",
                "_next": null,
                "_onEnter": null,
                "_onExit": null,
                "_open": true,
                "_parent": [Circular],
                "_prev": null,
                "_sourcepos": undefined,
                "_string_content": null,
                "_title": null,
                "_type": "text",
              },
              "_level": 1,
              "_listData": {},
              "_literal": null,
              "_next": [Circular],
              "_onEnter": null,
              "_onExit": null,
              "_open": false,
              "_parent": Node {
                "_destination": null,
                "_fenceChar": null,
                "_fenceLength": 0,
                "_fenceOffset": null,
                "_firstChild": [Circular],
                "_info": null,
                "_isFenced": false,
                "_lastChild": [Circular],
                "_level": null,
                "_listData": {},
                "_literal": null,
                "_next": null,
                "_onEnter": null,
                "_onExit": null,
                "_open": false,
                "_parent": null,
                "_prev": null,
                "_sourcepos": [
                  [
                    1,
                    1,
                  ],
                  [
                    4,
                    39,
                  ],
                ],
                "_string_content": null,
                "_title": null,
                "_type": "document",
              },
              "_prev": null,
              "_sourcepos": [
                [
                  1,
                  1,
                ],
                [
                  1,
                  8,
                ],
              ],
              "_string_content": null,
              "_title": null,
              "_type": "heading",
            },
            "_sourcepos": [
              [
                3,
                1,
              ],
              [
                4,
                39,
              ],
            ],
            "_string_content": null,
            "_title": null,
            "_type": "paragraph",
          },
          "_prev": null,
          "_sourcepos": undefined,
          "_string_content": null,
          "_title": null,
          "_type": "text",
        },
        "_sourcepos": undefined,
        "_string_content": null,
        "_title": null,
        "_type": "softbreak",
      },
      "_sourcepos": undefined,
      "_string_content": null,
      "_title": "",
      "_type": "image",
    },
    "originalSrc": "./path/to/image-1.jpg",
  },
]
`;

exports[`blog-post should return the rendered post 1`] = `
"<!DOCTYPE html><html lang="en"><head> <title>Post 1 </title><meta name="viewport" content="width=device-width, initial-scale=1"/><meta name="description" content="willjohnston.tech"/><link rel="icon" href="https://wjt.sfo2.cdn.digitaloceanspaces.com/wjt_logo.ico"/></head><style>*,*:before,*:after{box-sizing:border-box}*{margin:0}body{line-height:1.5;-webkit-font-smoothing:antialiased}img,picture,video,canvas,svg{display:block;max-width:100%}input,button,textarea,select{font:inherit}p,h1,h2,h3,h4,h5,h6{overflow-wrap:break-word}p{text-wrap:pretty}h1,h2,h3,h4,h5,h6{text-wrap:balance}#root,#__next{isolation:isolate}html{padding:1rem;font-family:Roboto,sans-serif}.post{height:100%;border:solid thin black;padding:1rem}
</style><div class="post"><h1>Post 1</h1>
<p>This is the first post.
<img src="./path/to/image-1.jpg" alt="image$100x200" /></p>
</div></html>"
`;

exports[`blog-post should update image sources 1`] = `
"<!DOCTYPE html><html lang="en"><head> <title>Post 1 </title><meta name="viewport" content="width=device-width, initial-scale=1"/><meta name="description" content="willjohnston.tech"/><link rel="icon" href="https://wjt.sfo2.cdn.digitaloceanspaces.com/wjt_logo.ico"/></head><style>*,*:before,*:after{box-sizing:border-box}*{margin:0}body{line-height:1.5;-webkit-font-smoothing:antialiased}img,picture,video,canvas,svg{display:block;max-width:100%}input,button,textarea,select{font:inherit}p,h1,h2,h3,h4,h5,h6{overflow-wrap:break-word}p{text-wrap:pretty}h1,h2,h3,h4,h5,h6{text-wrap:balance}#root,#__next{isolation:isolate}html{padding:1rem;font-family:Roboto,sans-serif}.post{height:100%;border:solid thin black;padding:1rem}
</style><div class="post"><h1>Post 1</h1>
<p>This is the first post.
<img src="https://example.com/path/to/image-1.jpg" alt="image$100x200" /></p>
</div></html>"
`;
